### ISSUE TITLE
No Sermon Details Metabox on localhost

### ISSUE THREAD

**Issue by zSeriesGuy** [WP-for-Church/Sermon-Manager#180]
When installing Sermon Manager with WP site set to localhost, there is no Sermon Detail metaboxes when adding a new sermon. 
Just to be sure of the results, I set up a clean install of WP and did not install any other plugins or themes.
Is this a known issue or by design?

![capture](https://user-images.githubusercontent.com/12520869/38326934-0686097e-3815-11e8-8093-5e13151fde9d.PNG)

**Comment by nikola3244** [comment]
Hi, 

That seems like a bug to me, thank you for reporting it.

Could you please send a screenshot of options in "Screen Details"? (in the top right corner)

Also, we'll need a bit more details:
- Sermon Manager version
- WordPress version
- PHP version

What is powering the localhost server? LAMP/MAMP/XAMPP/WAMP/Local by Flywheel?

**Comment by zSeriesGuy** [comment]
Bitnami wampstack-7.1.15-0
PHP 7.1.15
Wordpress 4.9.5
SermonManager 2.12.5

![capture3](https://user-images.githubusercontent.com/12520869/38330660-2b6276b0-381f-11e8-81b1-b525c4f8fb61.PNG)

**Comment by robertmain** [comment]
I'm having a hard time reproducing this..

WP Version: 4.9.5
SM Version: 2.12.5
PHP: 7.1.9
OS: Win 10

![image](https://user-images.githubusercontent.com/537684/39411899-9fd79d98-4be0-11e8-8530-a581cc6a066f.png)

**Comment by zSeriesGuy** [comment]
I thought it might have something to do with php extensions not being activated. But I have gone thru and verified that I now have all the extensions enabled that I have enabled on my VPS.
I still don’t get the metabox.
But if you can’t reproduce it, then that confirms that the issue is on my end. I’ll try to determine what the problem is.
Thank you,
Jerry

**Comment by zSeriesGuy** [comment]
Getting nothing in the PHP errors log.
I deactivated all plugins and still don’t get metabox. So it shouldn’t be a conflict. I’m testing with the basic WP TwentySeventeen theme.
Attached is my phpinfo output if you want to compare to yours. I feel like some extension is not active.

**Comment by robertmain** [comment]
Sure, but I wasn't meaning your plugin conflicts/wordpress logs etc. I was meaning your actual PHP error log. 

![image](https://user-images.githubusercontent.com/537684/39433595-41a6ecf4-4c64-11e8-94f2-a18c52b88dbc.png)

**Comment by zSeriesGuy** [comment]
Attached is my  apache log. For the apache log, I stopped apache, cleared the log, started apache, then went into the Add New sermon.
For the php errors log, no log was created. I have error logging set in php.ini, including log_errors=on and display_errors=on. No errors were generated.

**Comment by robertmain** [comment]
There's nothing attached. I don't think the whole email => GitHub issue thing deals with attachments very well..you may need to put it on gist.github.com or paste it into the issue directly.

**Comment by zSeriesGuy** [comment]
Here’s the apache log.

[Mon Apr 30 11:04:58.671058 2018] [ssl:warn] [pid 10704:tid 580] AH01909: localhost:443:0 server certificate does NOT include an ID which matches the server name
[Mon Apr 30 11:04:58.894063 2018] [ssl:warn] [pid 10704:tid 580] AH01909: localhost:443:0 server certificate does NOT include an ID which matches the server name
[Mon Apr 30 11:04:58.921108 2018] [mpm_winnt:notice] [pid 10704:tid 580] AH00455: Apache/2.4.29 (Win64) OpenSSL/1.0.2n PHP/7.1.15 configured -- resuming normal operations
[Mon Apr 30 11:04:58.921108 2018] [mpm_winnt:notice] [pid 10704:tid 580] AH00456: Apache Lounge VC14 Server built: Nov  5 2017 12:14:48
[Mon Apr 30 11:04:58.921108 2018] [core:notice] [pid 10704:tid 580] AH00094: Command line: 'E:\\Bitnami\\wampstack-7.1.15-0\\apache2\\bin\\httpd.exe -d E:/Bitnami/wampstack-7.1.15-0/apache2 -f E:\\Bitnami\\wampstack-7.1.15-0\\apache2\\conf\\httpd.conf'
[Mon Apr 30 11:04:58.927063 2018] [mpm_winnt:notice] [pid 10704:tid 580] AH00418: Parent: Created child process 13352
[Mon Apr 30 11:04:59.426093 2018] [ssl:warn] [pid 13352:tid 556] AH01909: localhost:443:0 server certificate does NOT include an ID which matches the server name
[Mon Apr 30 11:04:59.635099 2018] [ssl:warn] [pid 13352:tid 556] AH01909: localhost:443:0 server certificate does NOT include an ID which matches the server name
[Mon Apr 30 11:04:59.655100 2018] [mpm_winnt:notice] [pid 13352:tid 556] AH00354: Child: Starting 64 worker threads.

**Comment by robertmain** [comment]
Hmm, there's nothing in there that points to anything obviously wrong. Though, if you want to get rid of that error message - check [this](https://community.bitnami.com/t/apache-error-server-certificate-does-not-include-an-id-which-matches-the-server-name/52679/10) out

Here's what I'd try:

- Try replacing the plugin on disk (located in `wp-content/plugins/sermon-manager-for-wordpress`) with a fresh copy off `master`
- Try and get hold of a *NIX box and install a clean WP install and the plugin on that
- Give [this](https://digwp.com/2009/07/monitor-php-errors-wordpress/) a shot and see if anything interesting shows up in the wordpress logs or PHP error logs

**Comment by zSeriesGuy** [comment]
Sorry to take so long to respond on this.
I made the change to the wp-config.php to set wp_debug true.
Just in case, I restarted my apache server.
Went to Sermons->Add New.

There is no debugging information.
I also deactivated, delete, re-installed Sermon Manager.

All I see is the Add new sermon title box, the Discussion box, and the metadata boxes on the right for Preachers, Series, Topics.
But the blocks for specifying the other information is not there.

[cid:image002.jpg@01D3E50F.AA0EE790]

**Comment by robertmain** [comment]
That's weird. Just to double check, they're definitely enabled in "screen
options" at the top, right?

**Comment by zSeriesGuy** [comment]
Yes, all enabled.
This only seems to be happening on localhost. My Linux ubuntu VPS server is not having this issue.
My localhost development server is on Windows 10 Pro running bitnami wampstack 7.1.15.0.

I still think it has something to do with the php extensions. But then on the other hand, if it was, seems like I would get some kind of debug message about something not being available.

I have Nusphere PHPed. I can try running a profile and try tracing through some module if you want to point me to what module it would be that would generate those boxes.
Would it be sm-meta-box-functions.php?

[cid:image002.jpg@01D3E518.6CF815D0]

**Comment by nikola3244** [comment]
Hiya there, nice debugging - both of you! But this is still a very mysterious issue.

@zSeriesGuy `sm-meta-box-functions.php` does not exist anymore, so please download and install the latest development package (from [GitHub](https://github.com/WP-for-Church/Sermon-Manager/archive/dev.zip), or from [WordPress](https://downloads.wordpress.org/plugin/sermon-manager-for-wordpress.zip)).

The file that initializes the meta boxes is located in [`~/includes/admin/sm-cmb-functions.php`](https://github.com/WP-for-Church/Sermon-Manager/blob/dev/includes/admin/sm-cmb-functions.php), and the framework that renders them is located at [`~/includes/vendor/CMB2`](https://github.com/WP-for-Church/Sermon-Manager/tree/dev/includes/vendor/CMB2). I wonder if those files are loaded at all - let's test it:

1. Open `~/includes/admin/sm-cmb-functions.php`, for example
2. Replace `<?php` with `<?php exit;`
3. Save the file and reload the Sermons -> Add New screen

If the page is blank (just white), it means that the file is being loaded, but the issue is somewhere else. If the page is loading just fine, it means that the problem is deeper than that.

**Comment by zSeriesGuy** [comment]
Sorry to take so long getting back to this. I was on holiday and out of the country.

I did as you suggested and the page is all white.
The version of SM installed is 2.12.5.

**Comment by nikola3244** [comment]
Hey @zSeriesGuy, could you try opening the page in some other web browser?

Since you are using Windows 10, Edge is an option

**Comment by zSeriesGuy** [comment]
Using edge browser did not resolve it.

I believe I have gotten closer to the issue.

Sm-cmb-functions.php defines function wpfc_sermon_metaboxes and adds action cmb2_admin_init for that function.

Bootstrap.php defines function cmb2_bootstrap that invokes the cmb2_admin_init action.

Init.php at line 156 requires once bootstrap.php. And if the function exists, invokes cmb2_bootstrap which would then form the metaboxes.

When I trace step into the require_once for bootstrap.php, it is invoking /wampstack/frameworks/smarty/libs/bootstrap.php and not your bootstrap.php. It is never invoking your bootstrap.php.
Hence the cmb2_bootstrap function never gets invoked and no sermon metaboxes.

I don’t know anything about smarty. It appears to be some kind of template engine. https://www.smarty.net/

As a test, I made a copy of your bootstrap.php and called it xbootstrap.php and changed the require_once in init.php to xbootstrap.php.
THAT WORKED!
I then put init.php back to its original bootstrap.php and deleted the smarty directory from my wampstack frameworks.
It  works.

So apparently the autoloader system of my Bitnami Wampstack installation “prefers?” the smarty bootstrap.php over yours….perhaps some kind of naming conflict.
I did some google searches concerning the smarty bootstrap.php to see if there is some kind of naming requirement for apps to NOT use bootstrap.php when using smarty.
Haven’t really found anything yet. But I haven’t dug real deep.
For now, I will just not use smarty and my problem is resolved.

Jerry
